packages/next-swc/crates/next-core/src/app_source.rs

```diff
@@ -622,10 +676,138 @@ import {}, {{ chunks as {} }} from "COMPONENT_{}";
             Ok(())
         }
 
-        fn emit_metadata_warning(state: &mut State, files: &[FileSystemPathVc]) {
-            for file in files {
-                state.unsupported_metadata.push(*file);
+        fn write_metadata(state: &mut State, metadata: &Metadata) -> Result<()> {
+            if metadata.is_empty() {
+                return Ok(());
             }
+            let Metadata {
+                icon,
+                apple,
+                twitter,
+                open_graph,
+                favicon,
+                manifest,
+            } = metadata;
+            state.loader_tree_code += "  metadata: {";
+            write_metadata_items(state, "icon", favicon.iter().chain(icon.iter()))?;
+            write_metadata_items(state, "apple", apple.iter())?;
+            write_metadata_items(state, "twitter", twitter.iter())?;
+            write_metadata_items(state, "openGraph", open_graph.iter())?;
+            write_metadata_manifest(state, *manifest)?;
+            state.loader_tree_code += "  },";
+            Ok(())
+        }
+
+        fn write_metadata_manifest(
+            state: &mut State,
+            manifest: Option<MetadataItem>,
+        ) -> Result<()> {
+            let Some(manifest) = manifest else {
+                return Ok(());
+            };
+            match manifest {
+                MetadataItem::Static { path } => {
+                    use std::fmt::Write;
+                    let i = state.counter;
+                    state.counter += 1;
```

Comment:
`state.counter` shouldn't be exposed, there should be a `state.generate_identifier() -> (identifier, inner_module_id)` instead (that also pushes the import?). This could also be used for the alt metadata.